tests/integration/test_main.py

```diff
@@ -82,9 +83,69 @@ def test_empty_run(tmp_path):
     log_dir = './esmvaltool_output'
     log_file = os.path.join(log_dir,
                             os.listdir(log_dir)[0], 'run', 'main_log.txt')
+    filled_recipe = os.path.exists(
+        log_dir + '/' + os.listdir(log_dir)[0] + '/run/recipe_filled.yml')
     os.system("rm -r esmvaltool_output")
 
     assert log_file
+    assert not filled_recipe
+
+
+def test_filled_recipe(tmp_path, patched_datafinder):
+    """Test filled recipe is written."""
+    recipe_file = tmp_path / "recipe.yml"
+    content = dedent("""
+        documentation:
+          description: This is a test recipe.
+          authors:
+            - andela_bouwe
+          references:
+            - contact_authors
+            - acknow_project
+          projects:
+            - c3s-magic
+
+        datasets:
+          - dataset: bcc-csm1-1
+
+        preprocessors:
+          preprocessor_name:
+            extract_levels:
+              levels: 85000
+              scheme: nearest
+
+        diagnostics:
+          diagnostic_name:
+            additional_datasets:
+              - dataset: GFDL-ESM2G
+            variables:
+              ta:
+                preprocessor: preprocessor_name
+                project: CMIP5
+                mip: Amon
+                exp: historical
+                ensemble: r1i1p1
+                timerange: '*'
+                additional_datasets:
+                  - dataset: MPI-ESM-LR
+                    timerange: '*/P1Y'
+            scripts: null
+        """)
+    recipe_file.write_text(content)
+    Config.get_config_user(path=tmp_path)
+    excs = (ValueError, iris.exceptions.CoordinateNotFoundError)
```

Comment:
This seems a slightly odd choice of exceptions. I guess the `iris.exceptions.CoordinateNotFoundError` comes from the extract_levels preprocessor check that tries to see if there are pressure levels in the data. Might as well not use that preprocessor in the recipe. What causes the ValueError?

Reply:
I think it was something in the cmor checker

Reply:
OK, I guess it will fail to process the fake data at some point. You could have a look at the sample_data tests, these use real CMIP6 data from the [ESMValTool_sample_data](https://pypi.org/project/ESMValTool-sample-data/) package so they should be able to run the recipe without error, but this is fine too. I would simplify the test by removing the extract_levels preprocessor though, just the default preprocessor tests this feature well enough.

Reply:
Can iris.exceptions.CoordinateNotFoundError be removed now?

Reply:
Note that you can just pass `output_dir=str(tmp_path / 'esmvaltool_output')` as an argument to `ESMValTool.run`, but updating the config file like it is done now is fine too.